CHANGELOG.txt

```diff
@@ -2,6 +2,12 @@
 MABWiser CHANGELOG
 =====================
 
+-------------------------------------------------------------------------------
+March, 2, 2022 2.2.0
+-------------------------------------------------------------------------------
+major:
+- Modified `predict_expectations`, such that `predict` can use `predict_expectations` directly in all non-contextual learning policies
```

Comment:
I am having a bit of a hard time understanding the contract. 

Is this the invariant now? That is "for any mab model, predict() calls predict_expectation() internally and then returns the decision? 

What happens to arm_to_expectations. Can I still access it, and have a guarantee that these values are correctly calculated/stored? 

Sepearate question, on which LP policies, predict OR predict_exp() is not deterministic? That is, I call it twice, I get a different result? Is this documented/shared with the user?

Reply:
All the non-contextual learning policies predict() calls predict_expectations() and then returns a decision. The implementation for [Linear](https://github.com/fidelity/mabwiser/blob/master/mabwiser/linear.py#L200) is a bit different, but it fundamentally does the same.

`arm_to_expectations` is still available and its definition has not changed.

Re: determinism - In the PR the only policies with deterministic predict() and predict_exp() output are LinUCB and UCB1. In the current version only LinUCB and UCB1 have deterministic predict(), whereas all the LP policies except TS and LinTS have deterministic predict_expectations().

Reply:
Gotcha, looks like determinism is off track either way.. that cannot be helped I guess. Doruk's paper should shed more light on this :D 

Re: arm_to_exp, good to hear it still is there.